generator/src/main/java/org/apache/kafka/message/ApiMessageTypeGenerator.java

```diff
@@ -245,6 +247,58 @@ private void generateToString() {
         buffer.printf("}%n");
     }
 
+    private void generateHeaderVersion() {
+        buffer.printf("public short headerVersion(short version) {%n");
+        buffer.incrementIndent();
+        buffer.printf("switch (apiKey) {%n");
+        buffer.incrementIndent();
+        for (Map.Entry<Short, ApiData> entry : apis.entrySet()) {
+            short apiKey = entry.getKey();
+            buffer.printf("case %d:%n", apiKey);
+            buffer.incrementIndent();
+            if (apiKey == 7) {
```

Comment:
I think this one deserve a small comment to explain the special case. It wasn't obvious to me as least.

Reply:
added